analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -136,6 +144,43 @@ private bool IsEnabled(AnalyzerOptions options)
             return SupportedDiagnostics.Any(d => analyzerConfiguration.IsEnabled(d.Id));
         }
 
+        private void CheckWebConfig(SonarAnalysisContext context, CompilationAnalysisContext c)
+        {
+            foreach (var fullPath in context.GetWebConfig(c))
+            {
+                var webConfig = File.ReadAllText(fullPath);
+                if (webConfig.Contains("<system.web") && XmlHelper.ParseXDocument(webConfig) is { } doc)
+                {
+                    ReportRequestLengthViolation(doc, fullPath, c);
+                }
+            }
+        }
+
+        private void ReportRequestLengthViolation(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var httpRuntime in doc.XPathSelectElements("configuration/system.web/httpRuntime"))
+            {
+                if (httpRuntime.Attribute("maxRequestLength") is { } maxRequestLength
+                    && IsVulnerable(maxRequestLength.Value, MaxAllowedRequestLength)
+                    && maxRequestLength.CreateLocation(webConfigPath) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+            foreach (var requestLimit in doc.XPathSelectElements("configuration/system.webServer/security/requestFiltering/requestLimits"))
+            {
+                if (requestLimit.Attribute("maxAllowedContentLength") is { } maxAllowedContentLength
+                    && IsVulnerable(maxAllowedContentLength.Value, MaxAllowedContentLength)
+                    && maxAllowedContentLength.CreateLocation(webConfigPath) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+        }
+
+        private static bool IsVulnerable(string value, int limit) =>
```

Comment:
if it fails to parse, you consider it vulnerable ("hotspot")? 

It's not necessarily a bad thing, I guess (it's a hotspot, not a vulnerability), but can a webapp truly work with non-integer values? and if it doesn't, should we raise a security hotspot if there's actually a bug there?

Also, I see that for this scenario you added `Noncompliant FP` - so why keep it like this?